runtime/tracing/tracing-core/common/src/aws/smithy/kotlin/runtime/tracing/LoggingTraceProbe.kt

```diff
@@ -7,36 +7,41 @@ package aws.smithy.kotlin.runtime.tracing
 import mu.KLogger
 import mu.KotlinLogging
 
+private fun EventLevel.loggerMethod(): (KLogger, () -> Any?) -> Unit = when (this) {
+    EventLevel.Fatal,
+    EventLevel.Error,
+    -> KLogger::error
+    EventLevel.Warning -> KLogger::warn
+    EventLevel.Info -> KLogger::info
+    EventLevel.Debug -> KLogger::debug
+    EventLevel.Trace -> KLogger::trace
+}
+
+private fun EventLevel.throwableLoggerMethod(): (KLogger, Throwable, () -> Any?) -> Unit = when (this) {
+    EventLevel.Fatal,
+    EventLevel.Error,
+    -> KLogger::error
+    EventLevel.Warning -> KLogger::warn
+    EventLevel.Info -> KLogger::info
+    EventLevel.Debug -> KLogger::debug
+    EventLevel.Trace -> KLogger::trace
+}
+
 /**
  * A [TraceProbe] that sends logs events to downstream logging libraries (e.g., Slf4j on JVM). Those downstream
  * libraries may require configuration before messages will actually appear in logs.
  */
 public object LoggingTraceProbe : TraceProbe {
-    private fun EventLevel.loggerMethod(): (KLogger, () -> Any?) -> Unit = when (this) {
-        EventLevel.Fatal,
-        EventLevel.Error,
-        -> KLogger::error
-        EventLevel.Warning -> KLogger::warn
-        EventLevel.Info -> KLogger::info
-        EventLevel.Debug -> KLogger::debug
-        EventLevel.Trace -> KLogger::trace
-    }
-
-    private fun log(spanId: String, event: TraceEvent) {
+    private fun log(spanId: String, event: TraceEvent, message: TraceEventData.Message) {
         val logger = KotlinLogging.logger(event.sourceComponent)
-        val method = event.level.loggerMethod()
-        method(logger) {
-            val message = event.data as TraceEventData.Message
-
-            val content = message.content()
-            val exception = message.exception?.let { ": $it" } ?: ""
-
-            "$spanId: $content$exception"
+        when (message.exception) {
+            null -> event.level.loggerMethod()(logger) { "$spanId: ${message.content()}" }
+            else -> event.level.throwableLoggerMethod()(logger, message.exception) { "$spanId: ${message.content()}" }
         }
     }
 
     override fun postEvent(span: TraceSpan, event: TraceEvent) {
-        if (event.data is TraceEventData.Message) log(span.hierarchicalId, event)
+        if (event.data is TraceEventData.Message) log(span.hierarchicalId, event, event.data)
```

Comment:
**nit**: It seems slightly odd to do the unwrap and pass `event` and `event.data` given that `event` contains the data already which in theory makes the extra parameter redundant/unnecessary.

Reply:
It _is_ slightly odd, I agree, but I did it here to take advantage of smart-casting by the compiler. Since `postEvent` already knows that `event.data is TraceEventData.Message`, we can skip the re-casting in `log`.